Documentation for sequentialfile.

cczhu opened the pull request:
Added a tutorial for sequentialfile module, and cleaned up some of the associated docstrings.  Also added tutorial text pointing to the DADAFileNameSequencer, and cleaned up its docstring.

Addresses #46.

mhvk commented on docs/baseband/helpers/index.rst:
```diff
@@ -8,6 +10,92 @@ Helpers assist with reading and writing all file formats.  Currently,
+frameset of the sample file::
+
+    >>> fsf = vdif.open(filenames[1], mode='rs', sample_rate=fh.sample_rate)
+    >>> dps = fh.fh_raw.seek(40256)    # Seek to start of second frameset.
```
I think it is a bit confusing to use `fh_raw` here. I would just do
```
fh.seek(fh.size // 2)
<include output>
fsf.time0 == fh.time
True
np.all(fsf.read() == fh.read())
True
```

mhvk commented on docs/baseband/helpers/index.rst:
```diff
@@ -8,6 +10,92 @@ Helpers assist with reading and writing all file formats.  Currently,
+    >>> fh.close()  # Close sample file.
+
+While `~baseband.helpers.sequentialfile` can be used for any format, since
+file sequences are common for DADA, it is **implicitly used** if a list of
```
No need to use the ** here, I think.

mhvk commented on docs/baseband/dada/index.rst:
```diff
@@ -88,9 +90,18 @@ even smaller size of the payload, to show how one can define multiple files.
+write to them as a single contiguous file.  If, as above, one passes a template
+string, `~baseband.dada.open` uses `~baseband.dada.base.DADAFileNameSequencer`
+to create a subscriptable filename generator, which is then passed to
+|sequentialfile.open|.  For further details, and usage examples, see the
```
A more general question (for your overall docstring review): I'm not sure I find these "for further details, see ... API" sentences very useful; you've just linked to both `open` and `DADAFileNameSequencer`, so why repeat that? I ask in part since docstrings often get displayed in terminals, so there is some benefit in having to scroll less in being brief and to the point.

cczhu replied:
This was due to Nikhil's suggestion a while back to have all the possible arguments to `open` easily accessible in the readthedocs.  I agree it's clunky and redundant, since we're now linking to the API throughout the tutorials.  I'll mention that in general one should look through the API somewhere in the Getting Started docs, and then remove all of the additional mentions in the format docs.

cczhu replied:
Although in this specific case, I'm going to shorten rather than remove the sentence, since the tutorial for using `DADAFileNameSequencer` is in its docstring.

mhvk replied:
But you already link to `DADAFileNameSequencer` right above, so one can just click on that!

cczhu replied:
I'm just not sure it's obvious from the paragraph that you can click on it for more information about template strings.

cczhu commented:
Addressed PR comments, notably removing explicit referrals to the API throughout the documentation.  (Upon reflection, having that stated a lot wouldn't have helped me better understand Numpy, or SciPy).
